Angular renderer for object arrays

AlexandraBuzila opened the pull request:
Add array layout renderer for objects.

eneufeld commented on packages/angular-material/src/layouts/array-layout.renderer.ts:
```diff
@@ -0,0 +1,165 @@
+    `,
+    changeDetection: ChangeDetectionStrategy.OnPush
+  })
+  export class ArrayLayoutRenderer extends JsonFormsBaseRenderer<ControlElement>
```
why not extend the JsonFormsArrayControl?

AlexandraBuzila replied:
Good point! I changed the renderer so it extends `JsonFormsArrayControl`

eneufeld commented on packages/angular-material/src/layouts/array-layout.renderer.ts:
```diff
@@ -0,0 +1,165 @@
+            mat-icon-button
+            [disabled]="readonly"
+            (click)="add()"
+            attr.aria-label="{{ 'Add to ' + this.label + 'button' }}"
```
we will need to find a localization solution

eneufeld commented on packages/angular-material/src/layouts/array-layout.renderer.ts:
```diff
@@ -0,0 +1,145 @@
+
+  getProps(index: number): OwnPropsOfRenderer {
+    const uischema =
+      (this.uischema.options?.detailUISchema as UISchemaElement) ??
```
we should unify this with the react renderer and use `detail`

eneufeld replied:
or even better use the findschema like here: https://github.com/eclipsesource/jsonforms/blob/6cf39efbb82fdf77b173c38ca4b706d526920f9b/packages/material/src/layouts/ExpandPanelRenderer.tsx#L101

eneufeld commented on packages/angular-material/src/layouts/array-layout.renderer.ts:
```diff
@@ -0,0 +1,148 @@
+
+export const ArrayLayoutRendererTester: RankedTester = rankWith(
+  4,
+  isObjectArray
```
shouldn't we check for `isObjectArrayWithNesting` instead?

AlexandraBuzila replied:
Isn't that tester too restrictive? If we don't have a nested array or don't provide a detail schema in the options, then the new renderer won't be used.

eneufeld commented on packages/angular-material/src/layouts/array-layout.renderer.ts:
```diff
@@ -0,0 +1,148 @@
+    uischema: UISchemaElement;
+  }[];
+
+  constructor(private jsonFormAngularService: JsonFormsAngularService) {
```
we don't need to make this private as we can access it from the parent

eneufeld replied:
If not, it should be accessible via the parent

AlexandraBuzila replied:
Fixed

eneufeld commented:
Do you know why neither the buttons nor the toolbar is using the default styling colors of material?

AlexandraBuzila commented:
I did the small refactoring the the array renderer.

Regarding the styling:
- the button style was not consistent with the other renderers, I fixed that
- the toolbar styling was a little different because we were not using the Roboto font.  I added it to the Angular Material examples, but I can revert this change if you don't agree with it.
The toolbar should look now like in the Angular Material documentation: https://material.angular.io/components/toolbar/overview

AlexandraBuzila commented:
Ah, and the toolbar color is not set to 'primary'. I'll update my commit to use it.
